sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/storage/GcsFileSystemTest.java

```diff
@@ -168,6 +172,37 @@ public void testRecursiveGlobExpansionFails() throws IOException {
     gcsFileSystem.expand(GcsPath.fromUri("gs://testbucket/test**"));
   }
 
+  @Test
+  public void testMatchNonGlobs() throws Exception {
+    List<StorageObjectOrIOException> items = new ArrayList<>();
+    // Files within the directory
+    items.add(StorageObjectOrIOException.create(
+        createStorageObject("gs://testbucket/testdirectory/file1name", 1L /* fileSize */)));
+    items.add(StorageObjectOrIOException.create(new FileNotFoundException()));
+    items.add(StorageObjectOrIOException.create(new IOException()));
+    items.add(StorageObjectOrIOException.create(
+        createStorageObject("gs://testbucket/testdirectory/file4name", 4L /* fileSize */)));
+
+    List<GcsPath> gcsPaths = ImmutableList.of(
+        GcsPath.fromUri("gs://testbucket/testdirectory/file1name"),
+        GcsPath.fromUri("gs://testbucket/testdirectory/file2name"),
+        GcsPath.fromUri("gs://testbucket/testdirectory/file3name"),
+        GcsPath.fromUri("gs://testbucket/testdirectory/file4name"));
+
+    when(mockGcsUtil.getObjects(eq(gcsPaths))).thenReturn(items);
+    List<MatchResult> matchResults = gcsFileSystem.matchNonGlobs(gcsPaths);
+
+    assertEquals(4, matchResults.size());
+    assertThat(
```

Comment:
added javadoc to matchNonGlob().

cannot refer MatchResults in getObjects(), because GcsUtil is now in java/core, and cannot depends on io.
Even after we move GcsUtil to io/gcp, I still think GcsUtil shouldn't reference MatchResult.